server/tinylicious/src/routes/ordering/documents.ts

```diff
@@ -72,5 +81,33 @@ export function create(storage: IDocumentStorage): Router {
 		);
 	});
 
+	/**
+	 * Passes on content to all clients in a collaboration session happening on the document via means of signal.
+	 */
+	router.post("/:tenantId/:id/broadcast-signal", (request, response) => {
+		const tenantId = getParam(request.params, "tenantId");
+		const documentId = getParam(request.params, "id");
+		const signalContent = getParam(request.body, "signalContent");
+		const documentP = storage.getDocument(
+			getParam(request.params, "tenantId"),
+			getParam(request.params, "id"),
+		);
+		documentP.then(
+			(_document: any) => {
+				try {
+					const signalRoom: IRoom = { tenantId, documentId };
+					const payload: IBroadcastSignalEventPayload = { signalRoom, signalContent };
+					collaborationSessionEventEmitter.emit("broadcast-signal", payload);
+					response.status(200).send("OK");
+				} catch (error) {
+					response.status(500).send(error);
+				}
+			},
+			(error) => {
+				response.status(400).json(error);
```

Comment:
Let's make this a 500; we don't know that erroring here is because of a bad request.
@znewton I've seen places where we deliberately break that rule and return 400 when other codes would be better. Just checking if you have any comments about returning 500 here.

Reply:
This is a new API so that's fine. We use 400 in those instances because old code depends on a 400 code.